api/events/src/main/java/io/opentelemetry/api/events/DefaultEventEmitter.java

```diff
@@ -17,6 +17,47 @@ static EventEmitter getInstance() {
     return INSTANCE;
   }
 
+  @Override
+  public void emit(long epochNanos, String eventName, Attributes attributes) {}
+
   @Override
   public void emit(String eventName, Attributes attributes) {}
+
+  @Override
+  public EventBuilder builder() {
```

Comment:
Why the multiple overloads of builder? The version with no arguments makes it easy to do the wrong thing (i.e. emit an event with no name). The version with just an `eventName` argument is technically fine, but is just syntactic sugar for `builder(String eventName, Attributes.empty())` and not worth it IMO. I think we have:

- A single entrypoint into the builder which accepts the required arguments: `builder(String eventName, Attributes attributes)
- Only have builder setters for other fields. I.e. no `EventBuilder#setAttributes`, `EventBuilder#setEventName` since those are required

Reply:
I like the flexibility of the overloads, but I can see you're reluctant to have too much new code.

I've removed the builder overloads and removed `setEventName` and `setAttributes`. I was thinking it could be nice to reuse a builder with different attributes sometimes, but we can always add that back later as the need arises. 👍🏻

Reply:
This does result now, however, in an `SdkEventBuilder` with a 5 argument constructor that only has one setter. Seems like it kinda defeats the purpose of having a builder that way.

Reply:
forward compatibility, I think we got burned previously by not introducing a builder from the beginning

Reply:
Yeah there will more setters on the builder in the future.

Reply:
That's cool. I can live with that. 😎